### PR TITLE
More fax logging

### PR DESCRIPTION
**Pull request by Deatherd**
## About the PR
Added logging for copying and printing from a file using fax.

## Why / Balance
My solution for issue [#25694](https://github.com/space-wizards/space-station-14/issues/25694), that takes into account the logic of fax operation.

## Technical details
Now an admin log is added when adding a fax task to copy or print from a file.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

File: Content.Server/Fax/FaxSystem.cs [hunk 1]
```diff
@@ -416,13 +416,20 @@ public void PrintFile(EntityUid uid, FaxMachineComponent component, FaxFileMessa
         component.SendTimeoutRemaining += component.SendTimeout;
 
         UpdateUserInterface(uid, component);
+
+        if (args.Session.AttachedEntity != null)
+            _adminLogger.Add(LogType.Chat, LogImpact.Low,
+                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} has added to {ToPrettyString(uid):tool} queue the job to print following text from file: {args.Content}");
+        else
+            _adminLogger.Add(LogType.Chat, LogImpact.Low,
+                $"Someone has added to {ToPrettyString(uid):tool} queue the job to print following text from file: {args.Content}");
```
Comment (metalgearsloth) [inline Content.Server/Fax/FaxSystem.cs:425]:
```suggestion

        if (args.Session.AttachedEntity != null)
            _adminLogger.Add(LogType.Chat, LogImpact.Low,
                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} added print job to {ToPrettyString(uid):tool} with text: {args.Content}");
        else
            _adminLogger.Add(LogType.Chat, LogImpact.Low,
                $"Someone added print job to {ToPrettyString(uid):tool} with text: {args.Content}");
```

File: Content.Server/Fax/FaxSystem.cs [hunk 2]
```diff
@@ -449,6 +456,10 @@ public void Copy(EntityUid uid, FaxMachineComponent? component = null)
         // will start immediately.
 
         UpdateUserInterface(uid, component);
+
+        if (args.Session.AttachedEntity != null)
+            _adminLogger.Add(LogType.Chat, LogImpact.Low,
+                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} has added to {ToPrettyString(uid):tool} queue a job to copy {ToPrettyString(component.PaperSlot.Item):subject}");
     }
```
Comment (metalgearsloth) [inline Content.Server/Fax/FaxSystem.cs:463]:
```suggestion

        if (args.Session.AttachedEntity != null)
            _adminLogger.Add(LogType.Chat, LogImpact.Low,
                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} added print job to {ToPrettyString(uid):tool} with text: {ToPrettyString(component.PaperSlot.Item):subject}");
    }
```

File: Content.Server/Fax/FaxSystem.cs [hunk 3]
```diff
@@ -416,13 +416,20 @@ public void PrintFile(EntityUid uid, FaxMachineComponent component, FaxFileMessa
         component.SendTimeoutRemaining += component.SendTimeout;
 
         UpdateUserInterface(uid, component);
+
+        if (args.Session.AttachedEntity != null)
+            _adminLogger.Add(LogType.Chat, LogImpact.Low,
+                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} added print job to {ToPrettyString(uid):tool} with text: {args.Content}");
+        else
+            _adminLogger.Add(LogType.Chat, LogImpact.Low,
+                $"Someone added print job to {ToPrettyString(uid):tool} with text: {args.Content}");
```
Comment (metalgearsloth) [inline Content.Server/Fax/FaxSystem.cs:425]:
```suggestion
        if (args.Session.AttachedEntity != null)
            _adminLogger.Add(LogType.Action, LogImpact.Low,
                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} added print job to {ToPrettyString(uid):tool} with text: {args.Content}");
        else
            _adminLogger.Add(LogType.Action, LogImpact.Low,
                $"Someone added print job to {ToPrettyString(uid):tool} with text: {args.Content}");
```

File: Content.Server/Fax/FaxSystem.cs [hunk 4]
```diff
@@ -449,6 +456,10 @@ public void Copy(EntityUid uid, FaxMachineComponent? component = null)
         // will start immediately.
 
         UpdateUserInterface(uid, component);
+
+        if (args.Session.AttachedEntity != null)
+            _adminLogger.Add(LogType.Chat, LogImpact.Low,
+                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} added print job to {ToPrettyString(uid):tool} with text: {ToPrettyString(component.PaperSlot.Item):subject}");
     }
```
Comment (metalgearsloth) [inline Content.Server/Fax/FaxSystem.cs:463]:
```suggestion

        if (args.Session.AttachedEntity != null)
            _adminLogger.Add(LogType.Action, LogImpact.Low,
                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} added copy job to {ToPrettyString(uid):tool} with text: {ToPrettyString(component.PaperSlot.Item):subject}");
    }
```

File: Content.Server/Fax/FaxSystem.cs [hunk 5]
```diff
@@ -449,6 +456,10 @@ public void Copy(EntityUid uid, FaxMachineComponent? component = null)
         // will start immediately.
 
         UpdateUserInterface(uid, component);
+
+        if (args.Session.AttachedEntity != null)
+            _adminLogger.Add(LogType.Action, LogImpact.Low,
+                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} added copy job to {ToPrettyString(uid):tool} with text: {ToPrettyString(component.PaperSlot.Item):subject}");
```
Comment (exincore) [inline Content.Server/Fax/FaxSystem.cs:462]: `ToPrettyString(component.PaperSlot.Item):subject` does not print the contents; `paper.Content` does.

File: Content.Server/Fax/FaxSystem.cs [hunk 6]
```diff
@@ -416,13 +416,20 @@ public void PrintFile(EntityUid uid, FaxMachineComponent component, FaxFileMessa
         component.SendTimeoutRemaining += component.SendTimeout;
 
         UpdateUserInterface(uid, component);
+
+        if (args.Session.AttachedEntity != null)
+            _adminLogger.Add(LogType.Action, LogImpact.Low,
+                $"{ToPrettyString(args.Session.AttachedEntity.Value):actor} added print job to {ToPrettyString(uid):tool} with text: {args.Content}");
+        else
+            _adminLogger.Add(LogType.Action, LogImpact.Low,
+                $"Someone added print job to {ToPrettyString(uid):tool} with text: {args.Content}");
```
Comment (exincore) [inline Content.Server/Fax/FaxSystem.cs:425]: This is misleading... the `uid` is of the fax machine the player is interacting with, not the receiver of the print job.

Comment (exincore) [reply Content.Server/Fax/FaxSystem.cs:425]: > Isn't the receiving fax machine for file prints always the one that the player is interacting with? The signature of functions like this one usually has the entity ID be the owner of the component that is being passed

Oh, you are absolutely right; I mixed up which method body I was looking at.

**Pull request merged**